### PR TITLE
fix(card-in-card): adjust card in card wrapper width on Collapsed grid mode

### PR DESCRIPTION
**Pull request by proeung**
### Related Ticket(s)

https://github.com/carbon-design-system/carbon-for-ibm-dotcom/issues/5485

### Description

- This PR fixes the grid alignment issue seen on the `Card Group with Card in Card` variant when the grid mode is changed to `collapsed`. 

### Changelog

- Just some CSS styling changes.

### Testing Instructions

- Visit the `Card Group with Card in Card` variant - `?path=/story/components-card-group--with-card-in-card`
- Adjust the `Grid Mode` knob to `Collapsed (1px)`
- Ensure that the card within the `Card in Card` component aligns with the Card Group items.
- Ensure that the border-bottom line is set correctly on all breakpoints.

**Current:**
68747470733a2f2f696d616765732e7a656e68756275736572636f6e74656e742e636f6d2f3562653039633361363130323930306336643136643939312f38626461323332332d666433632d343734392d623137662d653434303437396332663166

**Expected:**
Screen Shot 2021-03-15 at 12 39 11 PM

### CODE REVIEW

**Review by shixiedesign — APPROVED** [review]
Thank you Putra! Looks great!

**Comment by proeung** [comment]
@shixiedesign No prob! Thanks for catching this alignment bug.

@RaphaelAmadeu @jeffchew Please take a look at the code and let me know if the changes look good or whether there are things within the docs that I'm missing.

File: packages/web-components/src/components/card-group/__stories__/card-group.stories.ts [hunk 1]
```diff
@@ -139,7 +139,7 @@ const gridModes = {
 export const withCardInCard = ({ parameters }) => {
   const { cards, gridMode } = parameters?.props?.CardGroup ?? {};
   return html`
-    <dds-card-in-card href="https://example.com">
+    <dds-card-in-card grid-mode="${ifNonNull(gridMode)}" href="https://example.com">
```
Comment (RaphaelAmadeu) [inline packages/web-components/src/components/card-group/__stories__/card-group.stories.ts:142]:
```suggestion
    <dds-card-in-card grid-mode="${gridMode}" href="https://example.com">
```

Just a simple point, we don't actually need this function if we're using select for the knob.

**Review by RaphaelAmadeu — APPROVED** [review]
LGTM!

**Comment by jeffchew** [comment]
Hey @proeung , can you take a look at the web components ci-checks? It looks to be failing at the moment.

**Comment by proeung** [comment]
> Hey @proeung , can you take a look at the web components ci-checks? It looks to be failing at the moment.

@jeffchew Just fixed the CI check error (see - https://github.com/carbon-design-system/carbon-for-ibm-dotcom/pull/5494/commits/0603f1415e0872bb521a158f6b0a93a8b83de2dd). Please take a look again.

**Comment by photodow** [comment]
I see this PR is to fix grid alignment, so I don't want to distract too much from the original intent. Do we have an issue opened already for the rounded corners and extra border width on focus for this component?

<a href="https://user-images.githubusercontent.com/3793636/111667878-24d1fe80-87e3-11eb-9bb0-e55690d13336.png" target="_blank"></a>

**Pull request merged**